packages/labs/react/src/create-component.ts

```diff
@@ -4,6 +4,14 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
+const NODE_MODE = false;
+const global = NODE_MODE ? globalThis : window;
+
+const htmlElementShimNeeded = NODE_MODE && global.HTMLElement === undefined;
+if (htmlElementShimNeeded) {
```

Comment:
Note in the reactive-element case, we also add the following at the end of the module:

```ts
if (htmlElementShimNeeded) {
  delete (global as Partial<typeof global>).HTMLElement;
}
```

https://github.com/lit/lit/pull/3156/files#diff-305d9543c5dd0eef82d2583a84422cfa2a1786f1f466692485e6dac5eb7d43baR1501

The goal is to avoid polluting the global namespace when we don't need to, and the the reason we used the "add it to global and then remove it later" trick was because TypeScript had trouble understanding ternaries that switched between global `HTMLElement` and a local/inline one.

Reply:
Right, thank you for pointing this out. I'm unsure if that's possible here. `HTMLElement` actually needs to be present during runtime. The error isn't happening strictly at import time but when React tries to server render the component, calling the `render()` method. Perhaps it might be better to change the code within `render()` to remove the reference to `HTMLElement` instead. That'll likely cause a conflict with https://github.com/lit/lit/pull/3128 though.

Reply:
https://github.com/lit/lit/commit/fe450aa4766dbce30dc6abbdfef1b320835f4adf

Here's what that would look like. It does feel cleaner to go about it this way..

Reply:
Yeah, I like that idea!

Reply:
> I modified it a bit, just in case a user happens to bring in a DOM shim that does define `HTMLElement`. @aomarks could you take another look? Thanks!

Good call, LGTM!